src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,128 @@
-/**
- * Contains metadata relevant to a representation.
- */
-import { Quad } from 'rdf-js';
+import { Store } from 'n3';
+import { BlankNode, Literal, NamedNode, Quad, Term } from 'rdf-js';
+import { quad as createQuad, literal, namedNode } from '@rdfjs/data-model';
 
 /**
- * Metadata corresponding to a {@link Representation}.
+ * Stores the metadata triples and provides methods for easy access.
  */
-export interface RepresentationMetadata {
+export class RepresentationMetadata {
+  private store: Store;
+  private id: NamedNode | BlankNode;
+
   /**
-   * All metadata triples of the resource.
+   * @param identifier - Identifier of the resource relevant to this metadata.
+   *                     A blank node will be generated if none is provided.
+   *                     Strings will be converted to named nodes.
+   * @param quads - Quads to fill the metadata with.
    */
-  raw: Quad[];
+  public constructor(identifier?: NamedNode | BlankNode | string, quads?: Quad[]) {
+    this.store = new Store(quads);
+    if (identifier) {
+      if (typeof identifier === 'string') {
+        this.id = namedNode(identifier);
+      } else {
+        this.id = identifier;
+      }
+    } else {
+      this.id = this.store.createBlankNode();
+    }
+  }
+
   /**
-   * Optional metadata profiles.
+   * @returns All metadata quads.
    */
-  profiles?: string[];
+  public quads(): Quad[] {
+    return this.store.getQuads(null, null, null, null);
```

Comment:
This actually does not work, since the parameters are not optional in the typings (although perhaps they should be) and are defined as `RDF.Term | string | null`.

Reply:
Oh, that's not good, should also be `undefined` indeed.